crates/runtime/src/http/v1.rs

```diff
@@ -447,20 +452,83 @@ pub(crate) mod datasets {
         match df_read.refresh_table(&dataset.name).await {
             Ok(()) => (
                 status::StatusCode::CREATED,
-                Json(DatasetRefreshResponse {
+                Json(MessageResponse {
                     message: format!("Dataset refresh triggered for {dataset_name}."),
                 }),
             )
                 .into_response(),
             Err(err) => (
                 status::StatusCode::INTERNAL_SERVER_ERROR,
-                Json(DatasetRefreshResponse {
+                Json(MessageResponse {
                     message: format!("Failed to trigger refresh for {dataset_name}: {err}."),
                 }),
             )
                 .into_response(),
         }
     }
+
+    pub(crate) async fn acceleration(
+        Extension(app): Extension<Arc<RwLock<Option<App>>>>,
+        Extension(df): Extension<Arc<RwLock<DataFusion>>>,
+        Path(dataset_name): Path<String>,
+        Json(payload): Json<AccelerationRequest>,
+    ) -> Response {
+        
+        
+        let app_lock = app.read().await;
+        let Some(readable_app) = &*app_lock else {
+            return (status::StatusCode::INTERNAL_SERVER_ERROR).into_response();
+        };
+
+        let Some(dataset) = readable_app
+            .datasets
+            .iter()
+            .find(|d| d.name.to_lowercase() == dataset_name.to_lowercase())
+        else {
+            return (
+                status::StatusCode::NOT_FOUND,
+                Json(MessageResponse {
+                    message: format!("Dataset {dataset_name} not found"),
+                }),
+            )
+                .into_response();
+        };
+
+        let acceleration_enabled = dataset.acceleration.as_ref().is_some_and(|f| f.enabled);
+
+        if !acceleration_enabled {
```

Comment:
You should be able to patch that field regardless is acceleration is enabled/disabled.

Reply:
Updated, thank you